File: algo/sort/insertion/ins.py
arr = [0, 7, 2, 5, 3, 6, 4, 1, 8, 9]
length = len(arr)
i = 0
j = 0

def insertionSort(arr):
    for i in range(0, length, +1):
        for j in range(length-1, i, -1):
            if arr[j] < arr[j-1]:
                arr[j], arr[j-1] = arr[j-1], arr[j]
    return arr

print(insertionSort(arr))

File: algo/sort/insertion/ins.js
import { swap } from "../swap.js";

const arr = [0, 7, 2, 5, 3, 6, 4, 1, 8, 9];
const n = arr.length;
let i, j;

const insertionSort = (arr) => {
  for (i = 0; i < n; i++) {
    for (j = i; j > 0; j--) {
      if (arr[j] < arr[j - 1]) {
        swap(arr, j, j - 1);
      }
    }
  }
  return arr;
};

console.log(insertionSort(arr));
